### PR TITLE
Added dotnet-ef - the core tool of .NET Entity Framework

### PR DESCRIPTION
**Pull request by lostmsu**
https://learn.microsoft.com/en-us/ef/core/cli/dotnet

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by corngood** [comment]
The only problem I see here is that the commit messages are non-standard.  The new package should be `[package name]: init at [version]`, and the maintainer commit should probably be prefixed with the module name.  Take a look at the history of that file for guidance.

Same goes for the other PR you mentioned, but we might as well finish one of them and rebase the other to remove the maintainer commit.

**Comment by corngood** [comment]
Thanks!  I see there's a 9.0.0 version on nuget.org.  Are these tied to runtime / SDK versions?  I'm just trying to understand if we'd need both 8+9, and if not, when would we want to move to 9.

**Comment by corngood** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361198`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>dotnet-ef</li>
  </ul>
</details>

**Comment by lostmsu** [comment]
I tried to upgrade and even though my system does not have .NET 9 the 9.0.0 worked, so I bumped the version.

My original reasoning was to stay on the LTS side just in case.

**Review by corngood — APPROVED** [review]
Looks good, thanks.

File: pkgs/by-name/do/dotnet-ef/package.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ buildDotnetGlobalTool, lib }:
+
+buildDotnetGlobalTool {
+  pname = "dotnet-ef";
+  version = "9.0.0";
+
+  nugetHash = "sha256-/Ru/H2WXX/SCqF2s0M1DJkaw+6Nikm+ccrveqiOXggA=";
+
+  meta = with lib; {
```
Comment (steeleduncan) [inline pkgs/by-name/do/dotnet-ef/package.nix:9]: `with lib` is deprecated #208242

**Review by steeleduncan — CHANGES_REQUESTED** [review]

File: pkgs/by-name/do/dotnet-ef/package.nix (hunk 1 above)
Comment (lostmsu) [reply pkgs/by-name/do/dotnet-ef/package.nix]: Replaced with full qualification

**Pull request merged**